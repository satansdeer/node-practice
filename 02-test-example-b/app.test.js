const runCommand = require("./utils");
const fsExtra = require("fs-extra");
const dirTree = require("directory-tree");

const APP_PATH = require.resolve("./app");

describe("app", () => {
  afterEach(() => {
    fsExtra.emptyDirSync("./data");
  });

  it("outputs info about created files", async () => {
    const stdout = await runCommand(`${APP_PATH} ./data/A.txt ./data/B.txt`);
    expect(stdout).toMatchSnapshot();
  });

  it("creates specified files", async () => {
    await runCommand(`${APP_PATH} ./data/A.txt ./data/B.txt`);

    const tree = dirTree("./data");
    expect(tree).toMatchSnapshot();
  });
});
