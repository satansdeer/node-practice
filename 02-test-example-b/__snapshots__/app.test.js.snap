// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app creates specified files 1`] = `
Object {
  "children": Array [
    Object {
      "extension": ".txt",
      "name": "A.txt",
      "path": "data/A.txt",
      "size": 0,
      "type": "file",
    },
    Object {
      "extension": ".txt",
      "name": "B.txt",
      "path": "data/B.txt",
      "size": 0,
      "type": "file",
    },
  ],
  "name": "data",
  "path": "./data",
  "size": 0,
  "type": "directory",
}
`;

exports[`app outputs info about created files 1`] = `
"Create Files
createFile: ./data/A.txt
createFile: ./data/B.txt
done: ./data/A.txt
done: ./data/B.txt
All files created
"
`;
